### PR TITLE
Extract pretty printer to separate library.

### PR DESCRIPTION
**Pull request by asummers**
This will allow parallel work for e.g. the GSOC project, Wobserver, SASL logs, and others.

### CODE REVIEW

**Comment by asummers** [comment]
You had mentioned at one point about running into issues with it Dialyzing itself. Curious what those were, this was mostly straightforward, and I didn't seem to run into them.

File: mix.exs [hunk 1]
```diff
@@ -24,7 +24,7 @@ defmodule Dialyxir.Mixfile do
   end
 
   def application do
-    [mod: {Dialyxir, []}, applications: [:dialyzer, :crypto, :mix]]
+    [mod: {Dialyxir, []}, applications: [:dialyzer, :crypto, :mix, :erlex]]
```
Comment (jeremyjh) [inline mix.exs]: I think this is redundant; since 1.4 Elixir adds all deps to the application list unless they have `runtime: false`.

**Comment by jeremyjh** [comment]
Well the issue with dialyzing itself doesn't happen when you install the task as a global archive. I didn't expect to be able to continue doing that since archives do not include dependencies. I'm guessing it still loads Erlex from the local dependency in this case, so it continues to work for us. We will need to remove the instructions about installing `dialyxir` globally from the README though as that will not continue to work for people.

**Comment by asummers** [comment]
Hmmm I wonder if we should just keep those and just clone the steps to point to my repo too. Then people would continue to be able to use that strategy.

File: mix.exs (hunk 1 above)
Comment (asummers) [reply mix.exs]: I'll play with it and see what all I can purge from here.

**Comment by asummers** [comment]
Credo does something similar with having you install Bunt, so it's not completely without precedence. 

https://github.com/rrrene/credo#using-credo-as-stand-alone

File: mix.exs [hunk 2]
```diff
@@ -24,7 +24,7 @@ defmodule Dialyxir.Mixfile do
   end
 
   def application do
-    [mod: {Dialyxir, []}, applications: [:dialyzer, :crypto, :mix, :erlex]]
```
Comment (jeremyjh) [inline mix.exs]: I actually just meant the `:erlex` was redundant - the other apps should be there since they are not `deps`.

Comment (asummers) [reply mix.exs]: It seemed to start okay and didn't freak out when used.

Comment (jeremyjh) [reply mix.exs]: Yeah I don't think any of those actually depend on getting a start as they are today,  but I think its best to leave them in terms of OTP correctness.

Comment (asummers) [reply mix.exs]: Did some research and apparently that's supposed to be `extra_applications` for things like crypto.

**Comment by asummers** [comment]
@jeremyjh Gonna go ahead and merge this I think. It now has `extra_applications` where it used to have `applications`

**Comment by jeremyjh** [comment]
> It now has extra_applications where it used to have applications

Excellent, good catch.

**Pull request merged**